rad/src/reducers/median.rs

```diff
@@ -0,0 +1,287 @@
+                            found: item.radon_type_name(),
+                        })
+                    }
+                }
+            }
+
+            input_not_nan.sort();
+
+            if input_not_nan.is_empty() {
+                // This can happen if all elements are NaN
+                Err(RadError::ModeEmpty)
+            } else if input_not_nan.len() % 2 == 1 {
+                // Odd number of elements: take element at floor(N/2):
+                let median_pos = input_not_nan.len() / 2;
+                let median_elem = input_not_nan[median_pos].into_inner();
+
+                Ok(RadonTypes::Float(RadonFloat::from(median_elem)))
+            } else {
+                // Even number of elements: take average of element at (N/2 - 1) and N/2
+                let right_pos = input_not_nan.len() / 2;
+                let right_elem = input_not_nan[right_pos].into_inner();
+                let left_pos = right_pos - 1;
+                let left_elem = input_not_nan[left_pos].into_inner();
+
+                // Create new array to be able to use average::mean reducer
+                let rl = RadonArray::from(vec![
+                    RadonTypes::Float(RadonFloat::from(left_elem)),
+                    RadonTypes::Float(RadonFloat::from(right_elem)),
+                ]);
+                // MeanReturnPolicy only applies to integers, so this will actually return a float
+                mean(&rl, MeanReturnPolicy::RoundToInteger)
+            }
+        }
+        Some(RadonTypes::Integer(_)) => {
+            let sorted_input =
+                match array_operators::sort(input, &[], &mut ReportContext::default()) {
+                    Ok(RadonTypes::Array(arr)) => arr.value(),
+                    Ok(_different_type) => unreachable!(),
+                    Err(e) => return Err(e),
+                };
+
+            if sorted_input.is_empty() {
+                // This is unreachable
+                Err(RadError::ModeEmpty)
+            } else if sorted_input.len() % 2 == 1 {
+                // Odd number of elements: take element at floor(N/2):
+                let median_pos = sorted_input.len() / 2;
+
+                Ok(sorted_input[median_pos].clone())
+            } else {
+                // Even number of elements: take average of element at (N/2 - 1) and N/2
+                let right_pos = sorted_input.len() / 2;
+                let left_pos = right_pos - 1;
+
+                // Create new array to be able to use average::mean reducer
+                let rl = RadonArray::from(vec![
+                    sorted_input[left_pos].clone(),
+                    sorted_input[right_pos].clone(),
+                ]);
+                // RoundToInteger means that when the average is not an integer, it will be rounded to an
+                // integer. For example, the average of 1 and 2, which is 1.5, will be rounded to 1.
+                mean(&rl, MeanReturnPolicy::RoundToInteger)
```

Comment:
If it is a Float, it is neccessary to round to Integer?

Reply:
According to the documentation of `MeanReturnPolicy`, `RoundToInteger` actually means "keep same type as the input":

```rust
/// Different available policies regarding what to do with the resulting Float after applying the
/// average mean.
#[derive(Clone, Copy, PartialEq, Eq)]
pub enum MeanReturnPolicy {
    /// Pure `Fn(Array<T>) -> Float` behavior.
    ReturnFloat,
    /// Enforces `Fn(Array<T>) -> T` behavior (if needed).
    RoundToInteger,
}
```